### ISSUE TITLE
AWS X-Ray http section not populated

### ISSUE THREAD

**Issue by fracampit** [marius-klimantavicius/opentelemetry-dotnet-exporter-xray#1]
First of all, thanks for implementing this package, it is very useful.

## Problem Description

I am using it on a .NET 8 Razor Pages website deployed on AWS AppRunner and configured with OTEL.

I am using this helper class to configure OTEL:
```
using System.Diagnostics;
using OpenTelemetry.Exporter.XRay;
using OpenTelemetry.Metrics;
using OpenTelemetry.Resources;
using OpenTelemetry.Trace;

namespace ######.Website.Configuration;

public static class OpenTelemetryHelper
{
    private static readonly ActivitySource Source = new("#######.DistributedTracing", "1.0.0");

    public static void Configure(IServiceCollection services)
    {
        services.AddSingleton(Source);
        
        var isProduction = Environment.GetEnvironmentVariable("ASPNETCORE_ENVIRONMENT") == "Production";

        services.AddOpenTelemetry()
            .ConfigureResource(resourceBuilder =>
            {
                resourceBuilder
                    .AddService("########")
                    .AddTelemetrySdk();
            })
            .WithTracing(tracerProviderBuilder =>
            {
                tracerProviderBuilder
                    .AddAWSInstrumentation() // AWS SDK instrumentation
                    .AddAspNetCoreInstrumentation() // ASP.NET Core instrumentation
                    .AddEntityFrameworkCoreInstrumentation() // EF Core instrumentation
                    .AddSource(Source.Name) // Your custom ActivitySource
                    .SetSampler(new AlwaysOnSampler()) // Sample all traces
                    .AddXRayExporter(options =>
                    {
                        options.LogGroupNames = ["#######"];
                        options.GenerateTraceIds = true;
                    });
                if (!isProduction)
                {
                    tracerProviderBuilder.AddConsoleExporter();
                }
            })
            .WithMetrics(metricBuilder =>
            {
                metricBuilder
                    .AddAspNetCoreInstrumentation()
                    .AddHttpClientInstrumentation()
                    .AddRuntimeInstrumentation()
                    .AddProcessInstrumentation();
            });
    }
}
```

I can see the traces in X-Ray, but the http section of the traces is not there. The information about the requests and responses is included in the annotations:
```
{
    "Id": "1-674cb742-d36b1eaffa44eb75a2babdab",
    "Duration": 0.005,
    "LimitExceeded": false,
    "Segments": [
        {
            "Id": "cc7c9dacf1b36e20",
            "Document": {
                "id": "cc7c9dacf1b36e20",
                "name": "#######",
                "start_time": 1733080898.1109564,
                "trace_id": "1-674cb742-d36b1eaffa44eb75a2babdab",
                "end_time": 1733080898.116426,
                "fault": false,
                "error": false,
                "throttle": false,
                "aws": {
                    "xray": {
                        "auto_instrumentation": false,
                        "sdk_version": "1.10.0",
                        "sdk": "opentelemetry for dotnet"
                    },
                    "cloudwatch_logs": [
                        {
                            "log_group": "#######"
                        }
                    ]
                },
                "annotations": {
                    "activity_display_name": "GET #######/#######/{id}/{handler?}",
                    "url_query": "?name=Redacted",
                    "network_protocol_version": "1.1",
                    "http_route": "#######/#######/{id}/{handler?}",
                    "http_response_status_code": 200,
                    "otel_resource_service_instance_id": "abafc63f-951f-432e-8582-112602e0dc7f",
                    "activity_operation_name": "Microsoft.AspNetCore.Hosting.HttpRequestIn",
                    "otel_resource_telemetry_sdk_name": "opentelemetry",
                    "http_request_method": "GET",
                    "otel_resource_telemetry_sdk_language": "dotnet",
                    "server_address": "#######.#######.net",
                    "otel_resource_telemetry_sdk_version": "1.10.0",
                    "url_scheme": "http",
                    "user_agent_original": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Cypress/13.13.0 Chrome/118.0.5993.159 Electron/27.3.10 Safari/537.36",
                    "otel_resource_service_name": "#######",
                    "url_path": "/#######/#######/a07a5ecd-f9b9-4992-b814-f036aa1e25d1"
                },
                "subsegments": [ ... ]
    }
}
```

**Comment by fracampit** [comment]
I was able to fix the problem by adding some custom tags to the traces:

```
        services.AddOpenTelemetry()
            .ConfigureResource(resourceBuilder =>
            {
                resourceBuilder
                    .AddService("######")
                    .AddTelemetrySdk()
                    .AddAttributes(new Dictionary<string, object>
                    {
                        ["cloud.provider"] = "aws",
                        ["cloud.platform"] = "aws_app_runner"
                    });
            })
            .WithTracing(tracerProviderBuilder =>
            {
                tracerProviderBuilder
                    .AddAWSInstrumentation()
                    .AddAspNetCoreInstrumentation(o =>
                    {
                        o.EnrichWithHttpRequest = (activity, httpRequest) =>
                        {
                            activity.SetTag("http.method", httpRequest.Method);
                            activity.SetTag("http.url", $"{httpRequest.Scheme}://{httpRequest.Host}{httpRequest.Path}{httpRequest.QueryString}");
                            activity.SetTag("http.target", httpRequest.Path);
                            activity.SetTag("http.host", httpRequest.Host);
                            activity.SetTag("http.scheme", httpRequest.Scheme);
                            activity.SetTag("http.client_ip", httpRequest.HttpContext.Connection.RemoteIpAddress?.ToString());
                            activity.SetTag("http.user_agent", httpRequest.Headers.UserAgent.ToString());
                        };
                        o.EnrichWithHttpResponse = (activity, httpResponse) =>
                        {
                            activity.SetTag("http.status_code", httpResponse.StatusCode);
                            activity.SetTag("http.response_content_length", httpResponse.ContentLength);
                        };
                        o.EnrichWithException = (activity, exception) =>
                        {
                            var exceptionTags = new ActivityTagsCollection
                            {
                                { "exception.type", exception.GetType().FullName },
                                { "exception.message", exception.Message },
                                { "exception.stacktrace", exception.ToString() }
                            };
                            var exceptionEvent = new ActivityEvent("exception", DateTime.UtcNow, exceptionTags);
                            activity.AddEvent(exceptionEvent);

                            // Optionally set the Activity status to error
                            activity.SetStatus(ActivityStatusCode.Error, exception.Message);
                        };
                    })
                    .AddSource(Source.Name)
                    .SetSampler(new AlwaysOnSampler())
                    .AddXRayExporter(options =>
                    {
                        options.LogGroupNames = ["######"];
                        options.GenerateTraceIds = true;
                    });
                if (!isProduction)
                {
                    tracerProviderBuilder.AddConsoleExporter();
                }
            })
            .WithMetrics(metricBuilder =>
            {
                metricBuilder
                    .AddAspNetCoreInstrumentation()
                    .AddHttpClientInstrumentation()
                    .AddRuntimeInstrumentation()
                    .AddProcessInstrumentation();
            });
```

Would you expect the exporter to work without doing the above?

**Comment by marius-klimantavicius** [comment]
Hi, thank you for excellent bug report!

Regarding http section it seems that span (activity) tags were finalized and stable names are used in latest version of AddAspNetCoreInstrumentation.

I've ported the latest http changes from https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/awsxrayexporter and these should be available in 1.0.14 (though I don't have enough time to test it with a real ASP.NET Core app)
